File: surf-api-bukkit/surf-api-bukkit-server/src/main/java/dev/slne/surf/surfapi/bukkit/server/impl/nms/listener/packets/serverbound/CommandSuggestionPacketImpl.java
package dev.slne.surf.surfapi.bukkit.server.impl.nms.listener.packets.serverbound;

import dev.slne.surf.surfapi.bukkit.api.nms.listener.packets.serverbound.CommandSuggestionPacket;
import net.minecraft.network.protocol.game.ServerboundCommandSuggestionPacket;

public final class CommandSuggestionPacketImpl extends NmsServerboundPacketImpl<ServerboundCommandSuggestionPacket> implements
    CommandSuggestionPacket {

  public CommandSuggestionPacketImpl(ServerboundCommandSuggestionPacket nmsPacket) {
    super(nmsPacket);
  }

  @Override
  public int completionId() {
    return getNmsPacket().getId();
  }

  @Override
  public String command() {
    return getNmsPacket().getCommand();
  }
}

File: surf-api-bukkit/surf-api-bukkit-server/src/main/kotlin/dev/slne/surf/surfapi/bukkit/server/impl/nms/listener/packets/serverbound/CommandSuggestionPacketImpl.kt
package dev.slne.surf.surfapi.bukkit.server.impl.nms.listener.packets.serverbound

import dev.slne.surf.surfapi.bukkit.api.nms.NmsUseWithCaution
import dev.slne.surf.surfapi.bukkit.api.nms.listener.packets.serverbound.CommandSuggestionPacket
import net.minecraft.network.protocol.game.ServerboundCommandSuggestionPacket

@NmsUseWithCaution
class CommandSuggestionPacketImpl(nmsPacket: ServerboundCommandSuggestionPacket) :
    NmsServerboundPacketImpl<ServerboundCommandSuggestionPacket>(nmsPacket),
    CommandSuggestionPacket {
    override val completionId get() = nmsPacket.id
    override val command get() = nmsPacket.command
}
